### PR TITLE
Check that repository exits for installing pipeline

### LINKED ISSUE #1188: spk install pipeline log messages are inaccurate and validation for pipeline definition file is incorrect
**Describe the bug:**
Screen Shot 2020-03-17 at 12 27 52 PM
The above log only reflects that the PAT given is valid for the AzDo org, we could combine this log to reflect that.

Additionally: 

```
mitarng@mtarng:~/Workspace/spk/tests/spk-env/fabrikam2019$ spk project install-lifecycle-pipeline --org-name mitarng --devops-project feb18project --repo-url https://dev.azure.com/mitarng/
feb18project/_git/doesnotexist --pipeline-name lifecycle-pipeline2
info:    Attempting to authenticate with Azure DevOps organization 'mitarng' using PAT '************************************************lh6q'
info:    Successfully connected to Azure DevOps Git API!
info:    Fetched DevOps Client
info:    Attempting to create new pipeline: lifecycle-pipeline2 defined in repository:https://dev.azure.com/mitarng/feb18project/_git/doesnotexist, branch: master, filePath: hld-lifecycle.
yaml
info:    Creating pipeline for definition
info:    Created pipeline for lifecycle-pipeline2
info:    Pipeline ID: 197
```
in the above example `https://dev.azure.com/mitarng/feb18project/_git/doesnotexist` does not exist, but somehow spk is still attempting to create the pipeline. 

**Expected behavior:**
Input validations should work correctly.

### PR DESCRIPTION
**Pull request by edaena**
Resolves: spk install pipeline log messages are inaccurate and validation for pipeline definition file is incorrect.

Closes [#1188](https://github.com/microsoft/bedrock/issues/1188)

### CODE REVIEW

**Comment by mtarng** [comment]
Does this only need to be done for `spk project install-lifecycle-pipeline`? what about `spk service install-build-pipeline` or `spk hld install-manifest-pipeline`?

**Comment by edaena** [comment]
@mtarng added the check to the other commands.

File: src/lib/azdoClient.test.ts [hunk 1]
```diff
@@ -11,7 +11,9 @@ import {
   getTaskAgentApi,
   getWebApi,
   invalidateWebApi,
+  repositoryExists,
```
Comment (edaena) [inline src/lib/azdoClient.test.ts]: remove

File: src/lib/azdoClient.test.ts [hunk 2]
```diff
@@ -144,6 +145,66 @@ describe("test getBuildApi function", () => {
...
+        "myFile",
+        "master",
+        "my-repo",
+        accessOpts
+      );
+    } catch (err) {
+      hasError = true;
+    }
+    expect(hasError).toBe(true);
+  });
```
Comment (dennisseah) [inline src/lib/azdoClient.test.ts:196]:
```
    let hasError = false;

    try {
      await validateRepository(
        "my-project",
        "myFile",
        "master",
        "my-repo",
        accessOpts
      );
    } catch (err) {
      hasError = true;
    }
    expect(hasError).toBe(true);
```
can be just
```
    await expect(validateRepository(
      "my-project",
      "myFile",
      "master",
      "my-repo",
      accessOpts
    )).rejects.toThrow();
```

**Review by dennisseah — APPROVED** [review]

**Review by mtarng — APPROVED** [review]

**Pull request merged**